bin/git-bulk

```diff
@@ -14,15 +14,25 @@ allwsmode=false
 #
 usage() {
   echo 1>&2 "usage: git bulk [-g] ([-a]|[-w <ws-name>]) <git command>"
-  echo 1>&2 "       git bulk --addworkspace <ws-name> <ws-root-directory>"
+  echo 1>&2 "       git bulk --addworkspace <ws-name> <ws-root-directory> (--from <URL or file>)"
   echo 1>&2 "       git bulk --removeworkspace <ws-name>"
   echo 1>&2 "       git bulk --addcurrent <ws-name>"
   echo 1>&2 "       git bulk --purge"
   echo 1>&2 "       git bulk --listall"
 }
 
 # add another workspace to global git config
-function addworkspace { git config --global bulkworkspaces."$wsname" "$wsdir"; }
+function addworkspace {
+	git config --global bulkworkspaces."$wsname" "$wsdir";
+	if [ ! -z "$source" ]; then
+		if ! cd "$wsdir" 2>/dev/null; then echo "Path of workspace doesn't exist, make it first."; exit 1; fi
+		regex='http(s)?://|ssh://(.*@)?.*\.git|(.*@)?.*:.*\.git'
```

Comment:
Yeah, now I tested the SSL and HTTP format of all repo hosts I knew (github, gitlab, bitbucket) and they work without `.git` extension, so I'll omit that part and just check the the beginning of URL…

Reply:
problem solved. btw I didn't change the order if file and pattern check, because it doesn't make any difference to computer! and for human reader it is not too complicated yet, since the whole regex pattern is defined in one line and that's where the magic happens…